dist/index.js

```diff
@@ -3304,24 +3307,31 @@ function run() {
             const packageNameList = packageName.split(RegExp("\\s"));
             const rosWorkspaceName = "ros_ws";
             const rosWorkspaceDir = path.join(workspace, rosWorkspaceName);
-            const sourceRosBinaryInstallation = core.getInput("source-ros-binary-installation");
-            const sourceRosBinaryInstallationList = sourceRosBinaryInstallation
-                ? sourceRosBinaryInstallation.split(RegExp("\\s"))
-                : [];
+            const targetRos1Distro = core.getInput("target-ros1-distro");
+            const targetRos2Distro = core.getInput("target-ros2-distro");
             const vcsRepoFileUrlListAsString = core.getInput("vcs-repo-file-url") || "";
             const vcsRepoFileUrlList = vcsRepoFileUrlListAsString.split(RegExp("\\s"));
-            const vcsRepoFileUrlListNonEmpty = vcsRepoFileUrlList.filter(x => x != "");
-            const vcsRepoFileUrlListResolved = vcsRepoFileUrlListNonEmpty.map(x => resolveVcsRepoFileUrl(x));
+            const vcsRepoFileUrlListNonEmpty = vcsRepoFileUrlList.filter((x) => x != "");
+            const vcsRepoFileUrlListResolved = vcsRepoFileUrlListNonEmpty.map((x) => resolveVcsRepoFileUrl(x));
             const coverageIgnorePattern = core.getInput("coverage-ignore-pattern");
             let commandPrefix = "";
-            if (sourceRosBinaryInstallation) {
-                if (process.platform !== "linux") {
-                    core.setFailed("sourcing binary installation is only available on Linux");
+            if (process.platform == "linux") {
+                if (!targetRos1Distro && !targetRos2Distro) {
+                    core.setFailed("No ROS1 or ROS2 distribution target specified for the build.");
                     return;
                 }
-                for (let rosDistribution of sourceRosBinaryInstallationList) {
-                    commandPrefix += `source /opt/ros/${rosDistribution}/setup.sh && `;
+            }
+            if (targetRos1Distro) {
+                if (validROS1Distro.indexOf(targetRos1Distro) <= -1) {
+                    return false;
+                }
+                commandPrefix += `source /opt/ros/${targetRos1Distro}/setup.sh && `;
```

Comment:
_this_ command is specific to linux and needs to be `if`ed. You might as well group the `touch` with it as well.

```suggestion
  if (isLinux) {
          commandPrefix += `mkdir -p /opt/ros/${targetRos1Distro} && touch /opt/ros/${targetRos1Distro}/setup.sh} && source /opt/ros/${targetRos1Distro}/setup.sh && `;
          }
```

ditto below